src/parse/decl.rs

```diff
@@ -931,6 +932,20 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
                             next: prefix.map(Box::new),
                         })
                     } else {
+                        if let Some(keyword) = self.match_next(&Token::Keyword(Keyword::Static)) {
+                            if allow_abstract {
+                                // TODO: Add information for the compiler to know
+                                // to warn if `NULL` is passed into the function
+                            } else {
+                                return Err(SyntaxError(Locatable {
```

Comment:
This is a semantic error, not a syntactic one. The difference is that semantic errors are recoverable while syntax errors are not. Use `self.semantic_err` instead of returning an error from the function.